apps/comments-ui/src/actions.ts

```diff
@@ -222,19 +242,33 @@ async function likeComment({state, api, data: comment}: {state: EditableAppConte
                 ...c,
                 replies
             };
-        })
+        }),
+        commentLikeLoadingId: comment.id
     };
 }
 
-async function reportComment({api, data: comment}: {api: GhostApi, data: {id: string}}) {
-    await api.comments.report({comment});
+async function sendUnlikeToApi({api, data: comment}: {api: GhostApi, data: {id: string, commentsState: Comment[]}}) {
+    const commentsState = comment.commentsState;
 
-    return {};
+    try {
+        await api.comments.unlike({comment: {id: comment.id}});
+        return {
+            commentUnlikeLoading: null
+        };
```

Comment:
Do we need this? It feels clunky.

Can we keep `likeComment` as it was where it still awaits the API response but first dispatches an action that does an immediate state update? That way the component that dispatches `likeComment` can await as normal and manage the button disabling with a typical wrapper around an async action but the state still updates immediately in the background to show the like/unlike status

Reply:
Pseudo-code:

```
async function likeComment(id) {
    dispatchAction('updateCommentLikeState', {id, liked: true});
    try {
        await api.likeComment(id);
    } catch () {
        dispatchAction('updateCommentLikeState', {id, liked: false});
    }
}
```